RubeesEat/Controllers/BillController.cs

```diff
@@ -0,0 +1,97 @@
+using System.Globalization;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Mvc;
+using RubeesEat.Model;
+
+namespace RubeesEat.Controllers;
+
+[ApiController]
+[Route("/api/bills")]
+public class BillController(IBillRepository billRepository, IPersonRepository personRepository)
+    : ControllerBase
+{
+    [Route("")]
+    [HttpGet]
+    public IEnumerable<Bill> GetAll()
+    {
+        return billRepository.GetAll();
+    }
+
+    [Route("")]
+    [HttpPost]
+    public Results<BadRequest<string>, RedirectHttpResult> Create()
+    {
+        var form = HttpContext.Request.Form;
+        string? description = form["billDescription"];
+        CultureInfo culture = new CultureInfo("en-EN");
+        if (description == null)
+        {
+            return TypedResults.BadRequest("Description was empty.");
+        }
+
+        List<EntryLine> entryLines = [];
+        if (!decimal.TryParse(form["billAmount"], culture, out decimal totalAmount))
+        {
+            return TypedResults.BadRequest("Total amount must be a number.");
+        }
+        entryLines.Add(new EntryLine(personRepository.GetCurrentUser(), totalAmount));
+
+        for (var i = 0;; i++)
+        {
+            string? id = form["id" + i];
+
+            if (id == null)
+            {
+                break;
+            }
+
+            Person? person;
+            if (!Guid.TryParse((id), out var personGuid))
+            {
+                return TypedResults.BadRequest("Person Guid not valid");
+            }
+
+            if (!personRepository.TryGetById(personGuid, out person))
+            {
+                return TypedResults.BadRequest("Person not found");
+            }
+
+            var stringValues = form["amount" + i];
+
+            if (!decimal.TryParse(stringValues, culture,out var amount))
+            {
+                return TypedResults.BadRequest("Amount of a person must be a number.");
+            }
+            
+            entryLines.Add(new EntryLine(person, -amount));
+        }
+        
+        if (entryLines[0].Amount <= 0)
+        {
+            return TypedResults.BadRequest("First entry should always be positive");
+        }
+
+        for(int i = 1; i < entryLines.Count; i++)
+        {
+            if (entryLines[i].Amount >= 0)
+            {
+                return TypedResults.BadRequest("Entries after the first one should always be negative");
+            }
+        }
+
+        bool hasDuplicates = entryLines
+                             .Skip(1)
+                             .GroupBy(p => p.Person)
+                             .Any(g => g.Count() > 1);
+        if (hasDuplicates)
+        {
+            return TypedResults.BadRequest("Persons should not appear multiple times (other than the buyer)");
+        }
```

Comment:
use .Distinct()

Reply:
according to this https://learn.microsoft.com/de-de/dotnet/api/system.linq.enumerable.distinct?view=net-8.0
.Distinct only returns a list of distinct elements. It does not tell me if the list only has distinct elements

Reply:
After discussion with Ottorium, we will compare the length of the list from Distinct with the original list